test/test.cpp

```diff
@@ -238,6 +240,78 @@ TEST(fclayer, get_dims_throws_when_0dim) {
 
 // ==========================
 
+// ==========================
+// Element-wise layer tests
+
+template <typename T>
+T minus(const T &elem) {
```

Comment:
My meaning that all simple functions need to be exist into the EWLayer because its part of the layer.
For example we read Alexnet layer and every layer has simple function, but your realisation know about functions nothing - it's problem. That why all dependence of EWLayer need to be in EWLayer (not outside). In tests we can use string for control what function(tanh or relu) need to be used

Reply:
Changed, used string for control